File: test_data/code/java/1089-duplicate-zeros.java
class Solution {
    public void duplicateZeros(int[] arr) {
        int length = arr.length;
        int index = length - 1;
        
        for(int i = length - 1; i >= 0; i--) {
            if(arr[i] == 0) {
                if(i < length - 1) {
                    arr[index] = 0;
                    index--;
                }
                arr[index] = 0;
                index--;
            } else {
                arr[index] = arr[i];
                index--;
            }
        }
    }
}

File: test_data/code/python/1089-duplicate-zeros.py
class Solution:
    def duplicateZeros(self, arr: List[int]) -> None:
        """
        Do not return anything, modify arr in-place instead.
        """
        i = 0
        n = len(arr) - 1
        while i < n:
            if arr[i] == 0:
                for j in range(n, i, -1):
                    arr[j] = arr[j-1]
                arr[i+1] = 0
                i += 1
                n -= 1
            else:
                i += 1